<launch>

    
    <arg name="robot_name" default="qingzhou_0" />
    <arg name="plan_topic" default="/$(arg robot_name)/global_plan" /> 

    <group ns="$(arg robot_name)">
        <param name="/robot_name/map" value="/$(arg robot_name)/map" />
        <param name="/robot_name/odom" value="/$(arg robot_name)/odom" />
        <param name="/robot_name/path" value="/$(arg robot_name)/path_planning_node/cleaning_plan_nodehandle/cleaning_path" />
        
        <param name="/NextGoal/tolerance_goal" value="2.00" />

        <node pkg="qingzhou_nav" type="next_goal" respawn="true" name="next_goal" output="screen" >
        </node>
        
         <node pkg="qingzhou_nav" type="path_planning.py" name="path_planner" output="screen">
            <param name="robot_name" value="$(arg robot_name)" />
            <param name="plan_topic" value="$(arg plan_topic)" />
        </node>

        <node pkg="qingzhou_nav" type="path_tracking.py" name="path_tracker" output="screen">
            <param name="robot_name" value="$(arg robot_name)" />
            <param name="plan_topic" value="$(arg plan_topic)" />
        </node>
    </group>


   

    <group ns="qingzhou_1">
        <param name="/robot_name/map" value="/qingzhou_1/map" />
        <param name="/robot_name/odom" value="/qingzhou_1/odom" />
        <param name="/robot_name/path" value="/qingzhou_1/path_planning_node/cleaning_plan_nodehandle/cleaning_path" />
        
        <param name="/NextGoal/tolerance_goal" value="2.00" />

        <node pkg="qingzhou_nav" type="next_goal" respawn="true" name="next_goal" output="screen" >
        </node>
        
         <node pkg="qingzhou_nav" type="path_planning.py" name="path_planner" output="screen">
            <param name="robot_name" value="qingzhou_1" />
            <param name="plan_topic" value="/qingzhou_1/global_plan" />
        </node>

        <node pkg="qingzhou_nav" type="path_tracking.py" name="path_tracker" output="screen">
            <param name="robot_name" value="qingzhou_1" />
            <param name="plan_topic" value="/qingzhou_1/global_plan" />
        </node>
    </group>
    


       <group ns="qingzhou_2">
        <param name="/robot_name/map" value="/qingzhou_2/map" />
        <param name="/robot_name/odom" value="/qingzhou_2/odom" />
        <param name="/robot_name/path" value="/qingzhou_2/path_planning_node/cleaning_plan_nodehandle/cleaning_path" />
        
        <param name="/NextGoal/tolerance_goal" value="2.00" />

        <node pkg="qingzhou_nav" type="next_goal" respawn="true" name="next_goal" output="screen" >
        </node>
        
         <node pkg="qingzhou_nav" type="path_planning.py" name="path_planner" output="screen">
            <param name="robot_name" value="qingzhou_2" />
            <param name="plan_topic" value="/qingzhou_2/global_plan" />
        </node>

        <node pkg="qingzhou_nav" type="path_tracking.py" name="path_tracker" output="screen">
            <param name="robot_name" value="qingzhou_2" />
            <param name="plan_topic" value="/qingzhou_2/global_plan" />
        </node>
    </group>


     <group ns="qingzhou_3">
        <param name="/robot_name/map" value="/qingzhou_3/map" />
        <param name="/robot_name/odom" value="/qingzhou_3/odom" />
        <param name="/robot_name/path" value="/qingzhou_3/path_planning_node/cleaning_plan_nodehandle/cleaning_path" />
        
        <param name="/NextGoal/tolerance_goal" value="2.00" />

        <node pkg="qingzhou_nav" type="next_goal" respawn="true" name="next_goal" output="screen" >
        </node>
        
         <node pkg="qingzhou_nav" type="path_planning.py" name="path_planner" output="screen">
            <param name="robot_name" value="qingzhou_3" />
            <param name="plan_topic" value="/qingzhou_3/global_plan" />
        </node>

        <node pkg="qingzhou_nav" type="path_tracking.py" name="path_tracker" output="screen">
            <param name="robot_name" value="qingzhou_3" />
            <param name="plan_topic" value="/qingzhou_3/global_plan" />
        </node>
    </group>


  
</launch>


